### ISSUE TITLE
[6.0] resolverMainFields breaks uuid library / crypto polyfill

### ISSUE THREAD

**Issue by jd20** [storybookjs/react-native#255]
**Describe the bug**
Per #174, it's necessary to set `resolverMainFields: ['sbmodern', 'main']` to get SB 6.0 working with React Native. However, this seems to break projects that depend on `uuid` (probably due to the way `crypto`  module is polyfilled in the `index.js`, see https://github.com/uuidjs/uuid#react-native--expo).

Removing the `resolverMainFields` will fix the `uuid` error on app startup, but obviously that breaks other things in Storybook. I've also tried removing `uuid`, and found `nanoid` is another library with same issue, it's hard to avoid having any dependencies on either of these in a large project.

**To Reproduce**
1. Create a React Native app with SB 6.0.
2. `yarn add uuid react-native-get-random-values`
3. At the beginning of `index.js`, add `import 'react-native-get-random-values'`.
4. Anywhere in `App.tsx`, reference uuid library, for example: `import {v4} from 'uuid'; v4();`
5. Run `npx react-native run-ios`, see error on app startup.
6. Observe the error: `Unable to resolve module crypto from .../RnSBSixAlpha/node_modules/uuid/dist/rng.js: crypto could not be found within the project or in these directories: node_modules`
7. Removing the `resolverMainFields` override from `metro.config.js` will make the error go away.

Here is a link to a minimal example, that reproduces the issue: https://github.com/jd20/RnSBSixAlpha

**Expected behavior**
The app should load without errors.

**Screenshots**
<img width="520" alt="Screen Shot 2021-08-16 at 7 01 27 PM" src="https://user-images.githubusercontent.com/17896438/129651909-5a84a3df-a69d-48c7-a7ed-4c21ba615d11.png">

**System:**
```
  System:
    OS: macOS 11.4
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
  Binaries:
    Node: 15.5.1 - ~/.asdf/installs/nodejs/15.5.1/bin/node
    Yarn: 1.22.10 - ~/.asdf/shims/yarn
    npm: 7.3.0 - ~/.asdf/installs/nodejs/15.5.1/bin/npm
  Browsers:
    Chrome: 92.0.4515.131
    Safari: 14.1.1
  npmPackages:
    @storybook/addon-actions: ^6.3.7 => 6.3.7
    @storybook/addon-controls: ^6.3.7 => 6.3.7
    @storybook/addon-docs: ^6.3.7 => 6.3.7
    @storybook/addon-ondevice-actions: ^6.0.1-alpha.0 => 6.0.1-alpha.0
    @storybook/addon-ondevice-backgrounds: ^6.0.1-alpha.0 => 6.0.1-alpha.0
    @storybook/addon-ondevice-controls: ^6.0.1-alpha.0 => 6.0.1-alpha.0
    @storybook/addon-ondevice-notes: ^6.0.1-alpha.0 => 6.0.1-alpha.0
    @storybook/addon-storyshots: ^6.3.7 => 6.3.7
    @storybook/react-native: ^6.0.1-alpha.1 => 6.0.1-alpha.1
```

**Comment by dannyhw** [comment]
are you using expo?

can you try changing your resolverMainFields to extend the existing metro config instead of overwriting it?

it would be something like this, depends what your setup is though.
`resolverMainFields: ["sbmodern" , ...default.resolver.resolverMainFields]`

**Comment by jd20** [comment]
Yes, that works! We are not using Expo. Extending the default config was a bit complicated, here's what I did below, but a much easier fix might be to just set `resolverMainFields: ["sbmodern" , "browser", "main"]`.

```ts
const {getDefaultConfig} = require('metro-config');

module.exports = (async () => {
  const {
    resolver: {resolverMainFields},
  } = await getDefaultConfig();
  return {
    transformer: {
      getTransformOptions: async () => ({
        transform: {
          experimentalImportSupport: false,
          inlineRequires: false,
        },
      }),
    },
    resolver: {
      resolverMainFields: ['sbmodern', ...resolverMainFields],
    },
  };
})();
```

Would you like me to submit a PR to update the 6.0 instructions, with either of these approaches?

**Comment by dannyhw** [comment]
@jd20 great news! Yes please, It would be awesome If you could open a PR. Go with the solution you would suggest then we can discuss further in the PR, how does that sound?

**Comment by jd20** [comment]
Created #256 .

**Comment by dannyhw** [comment]
This was fixed by merging #256

**Comment by simon-nicholls** [comment]
Using the patched docs/script I still had an error resolving nanoid, but this worked for me:

`resolverMainFields: ['sbmodern', 'react-native', 'browser', 'main']`
